okta/services/idaas/resource_okta_network_zone.go

```diff
@@ -396,3 +436,66 @@ func mapNetworkZoneToState(d *schema.ResourceData, data *v5okta.ListNetworkZones
 	}
 	return err
 }
+
+// updateNetworkZoneWithExemptList makes a custom HTTP request to update a network zone
+// with the useAsExemptList field included in the JSON body
+func updateNetworkZoneWithExemptList(ctx context.Context, meta interface{}, zoneID string, payload v5okta.ListNetworkZones200ResponseInner) error {
+	// Get the configuration from meta
+	cfg := meta.(*config.Config)
+
+	// Build the URL (no query parameter needed)
+	// The cfg.Domain is just the base domain like "okta.com",
+	// but we need the full org URL like "https://trial-7001215.okta.com"
+	baseURL := fmt.Sprintf("https://%s.%s", cfg.OrgName, strings.TrimSuffix(cfg.Domain, "/"))
+	endpoint := fmt.Sprintf("/api/v1/zones/%s", zoneID)
+	fullURL := baseURL + endpoint
+
+	// Convert SDK payload to map for manipulation
+	jsonData, err := json.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("failed to marshal payload: %v", err)
+	}
+
+	// Parse JSON into a map so we can add the useAsExemptList field
+	var payloadMap map[string]interface{}
+	if err := json.Unmarshal(jsonData, &payloadMap); err != nil {
+		return fmt.Errorf("failed to unmarshal payload: %v", err)
+	}
+
+	// Add the useAsExemptList field to the JSON payload
+	payloadMap["useAsExemptList"] = true
+
+	// Re-marshal with the added field
+	finalJsonData, err := json.Marshal(payloadMap)
+	if err != nil {
+		return fmt.Errorf("failed to marshal final payload: %v", err)
+	}
+
+	// Create the HTTP request
+	req, err := http.NewRequestWithContext(ctx, "PUT", fullURL, bytes.NewBuffer(finalJsonData))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %v", err)
+	}
+
+	// Set required headers
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Authorization", "SSWS "+cfg.ApiToken)
+	req.Header.Set("User-Agent", "terraform-provider-okta")
+
+	// Make the request
+	httpClient := &http.Client{}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to make request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	// Check response status
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
When making partial updates, the Okta API can sometimes return a `2**` response and not take action, do we need to account for that here?

We can potentially account for this by returning the resource's `read()` function in the resource's `update()` function?